gstreamer 1.16.1

dtzWill opened the pull request:
###### Motivation for this change


###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

nh2 committed:
gst_all_1: Fix evaluation on Darwin, fix gstreamer builds.

Commit be382109ad01d0aa8660f2b19a20e345c86d246b broke evaluation
on Darwin because it added these unconditional buildInputs
that don't evaluate on Darwin:

    libnice librdf lilv lv2 serd sord sratom

This commit fixes it, and also fixes recently-added new dependencies
for other packages accordingly.

It further fixes the build of many gstreamer packages on Darwin.

dtzWill committed:
gst_all_1: pname-ification

dtzWill committed:
gst_all_1.gst-editing-services: 1.16.0 -> 1.16.1

dtzWill committed:
gst_all_1.gst-plugins-base: 1.16.0 -> 1.16.1

dtzWill committed:
gst_all_1.gst-rtsp-server: 1.16.0 -> 1.16.1

dtzWill committed:
gst_all_1.gstreamer: 1.16.0 -> 1.16.1

dtzWill committed:
gst_all_1.gst-libav: 1.16.0 -> 1.16.1

dtzWill committed:
gst_all_1.gst-plugins-good: 1.16.0 -> 1.16.1

dtzWill committed:
gst_all_1.gst-vaapi: 1.16.0 -> 1.16.1

dtzWill committed:
gst_all_1.gst-plugins-bad: 1.16.0 -> 1.16.1

dtzWill committed:
gst_all_1.gst-plugins-ugly: 1.16.0 -> 1.16.1

dtzWill committed:
gst_all_1.gst-validate: 1.16.0 -> 1.16.1

dtzWill commented:
Requesting reviews guided by GitHub's unmatched wisdom in suggested reviewers! ;)

jtojnar reviewed (changes requested):
According to [release notes](https://gstreamer.freedesktop.org/releases/1.16/#1.16.1):

* bad: should build with opencv4
* bad: Does not newly added `colormanagement` meson option break stuff?
* vaapi: the patch seems to have been backported to 1.16.1: https://gitlab.freedesktop.org/gstreamer/gstreamer-vaapi/commit/8c11de79260105c5ad38454c7766290e6994fa11

jtojnar committed:
libnice: clean up

* format withn nixpkgs-fmt
* drop unnecessary flags
* do not propagate gupnp-igd (it is only in Requires.private)

jtojnar commented:
After swithching to opencv4, I am getting the following warnings:

```
[54/844] Compiling C++ object 'gst-libs/gst/opencv/13174e1@@gstopencv-1.0@sha/gstopencvutils.cpp.o'.K-enumtypes.c.o'.symbols'..bols'.
cc1plus: warning: /nix/store/bwmrzfni2r4pq7mi1p39xi2dcy7g9af1-opencv-4.1.2/include/opencv4/opencv: No such file or directory [-Wmissing-include-dirs]
[64/844] Compiling C++ object 'gst-libs/gst/opencv/13174e1@@gstopencv-1.0@sha/gstopencvvideofilter.cpp.o'.ispatcher.c.o'.
cc1plus: warning: /nix/store/bwmrzfni2r4pq7mi1p39xi2dcy7g9af1-opencv-4.1.2/include/opencv4/opencv: No such file or directory [-Wmissing-include-dirs]
```

Though, they do not seem to affect build.

jtojnar commented:
Clean ups in a fixup commits. Trying to see if base build on Darwin with Meson now.

jtojnar committed:
libpsl: fix darwin build

tests are failing there

FAIL: test-is-public
====================
valgrind: mmap-FIXED(0x7fff5f400000, 8388608) failed in UME (load_unixthread1) with error 22 (Invalid argument).
FAIL test-is-public (exit status: 1)
FAIL: test-is-public-all
========================
valgrind: mmap-FIXED(0x7fff5f400000, 8388608) failed in UME (load_unixthread1) with error 22 (Invalid argument).
FAIL test-is-public-all (exit status: 1)
FAIL: test-is-cookie-domain-acceptable
======================================
valgrind: mmap-FIXED(0x7fff5f400000, 8388608) failed in UME (load_unixthread1) with error 22 (Invalid argument).
FAIL test-is-cookie-domain-acceptable (exit status: 1)
FAIL: test-is-public-builtin
============================
valgrind: mmap-FIXED(0x7fff5f400000, 8388608) failed in UME (load_unixthread1) with error 22 (Invalid argument).
FAIL test-is-public-builtin (exit status: 1)
FAIL: test-registrable-domain
=============================
valgrind: mmap-FIXED(0x7fff5f400000, 8388608) failed in UME (load_unixthread1) with error 22 (Invalid argument).
FAIL test-registrable-domain (exit status: 1)

nh2 commented:
If you are looking for Darwin fixes, consider the top commit of #60922.

worldofpeace commented:
@jtojnar Glorious cleanups :+1: Taking a look.

worldofpeace commented on pkgs/development/libraries/gstreamer/libav/default.nix:
```diff
@@ -1,43 +1,46 @@
+    ninja
+    gettext
+    pkgconfig
+    python
```
Is this python3?

worldofpeace commented on pkgs/development/libraries/gstreamer/vaapi/default.nix:
```diff
@@ -5,11 +5,11 @@
   src = fetchurl {
-    url = "${meta.homepage}/src/${pname}/${pname}-${version}.tar.xz";
-    sha256 = "07qpynamiz0lniqajcaijh3n7ixs4lfk9a5mfk50sng0dricwzsf";
+    url = "${meta.homepage}/src/gstreamer-vaapi/gstreamer-vaapi-${version}.tar.xz";
```
Was this renamed?

jtojnar replied:
This is just a partial revert of df63b16126a330f89e64aaa4cdd327cdeb0f406f.

worldofpeace commented on pkgs/development/libraries/gstreamer/vaapi/default.nix:
```diff
@@ -12,36 +27,41 @@ stdenv.mkDerivation rec {
+    homepage = "https://gstreamer.freedesktop.org";
     license = stdenv.lib.licenses.lgpl21Plus;
     platforms = stdenv.lib.platforms.linux;
     maintainers = with stdenv.lib.maintainers; [ tstrobel ];
```
Can we do `with stdenv.lib;`?

worldofpeace commented on pkgs/development/libraries/gstreamer/validate/default.nix:
```diff
@@ -21,19 +20,24 @@ stdenv.mkDerivation rec {
+    description = "Integration testing infrastructure for the GStreamer framework";
+    homepage = "https://gstreamer.freedesktop.org";
+    license = stdenv.lib.licenses.lgpl2Plus;
+    platforms = stdenv.lib.platforms.unix;
```
`with stdenv.lib;`?

worldofpeace commented on pkgs/development/libraries/libnice/default.nix:
```diff
@@ -21,34 +36,48 @@ stdenv.mkDerivation rec {
+
+  propagatedBuildInputs = [
+    glib
+    gupnp-igd
```
I'm seeing `gupnp-igd` in `Requires.private`.

worldofpeace commented on pkgs/development/libraries/gstreamer/ges/default.nix:
```diff
@@ -1,39 +1,63 @@
-  propagatedBuildInputs = [ gst-plugins-base libxml2 ];
+  propagatedBuildInputs = [
+    gst-plugins-base
+    libxml2
```
I don't see `libxml2` in `gst-editing-services-1.0.pc`.

jtojnar commented:
@GrahamcOfBorg build gst_all_1

jtojnar commented:
@GrahamcOfBorg build gst_all_1.gst-editing-services gst_all_1.gst-plugins-base gst_all_1.gst-rtsp-server gst_all_1.gstreamer gst_all_1.gst-libav gst_all_1.gst-plugins-good gst_all_1.gst-vaapi gst_all_1.gstreamermm gst_all_1.gst-plugins-bad gst_all_1.gst-plugins-ugly gst_all_1.gst-validate

jtojnar commented:
Hmm, good linking error on darwin:

```
[399/680] Linking target ext/aalib/libgstaasink.dylib.
FAILED: ext/aalib/libgstaasink.dylib
/nix/store/x7z7hy6k9dgq1lw02bvd73y9qvqnzfny-clang-wrapper-7.1.0/bin/cc  -o ext/aalib/libgstaasink.dylib 'ext/aalib/e5079de@@gstaasink@sha/gstaasink.c.o' -Wl,-dead_strip_dylibs -Wl,-headerpad_max_install_names -shared -install_name @rpath/libgstaasink.dylib /nix/store/lm8zs7vi27zpjf3inlpjksz7ydafsq6r-gst-plugins-base-1.16.1/lib/libgstvideo-1.0.dylib /nix/store/f20wqw4jb84vwnpi2ampwcy5pv9bvg13-gstreamer-1.16.1/lib/libgstbase-1.0.dylib /nix/store/f20wqw4jb84vwnpi2ampwcy5pv9bvg13-gstreamer-1.16.1/lib/libgstreamer-1.0.dylib /nix/store/7zgjfr8sw11dmcq7q4dm6r6c4mv1cppv-glib-2.62.2/lib/libgobject-2.0.dylib /nix/store/7zgjfr8sw11dmcq7q4dm6r6c4mv1cppv-glib-2.62.2/lib/libglib-2.0.dylib -laa
Undefined symbols for architecture x86_64:
  "_cbreak", referenced from:
      _curses_init in libaa.a(aacurkbd.o)
  "_curs_set", referenced from:
      _curses_cursor in libaa.a(aacurses.o)
  "_echo", referenced from:
      _curses_uninit in libaa.a(aacurkbd.o)
  "_endwin", referenced from:
      _curses_uninit in libaa.a(aacurses.o)
      _curses_getsize in libaa.a(aacurses.o)
      _curses_uninit in libaa.a(aacurkbd.o)
  "_initscr", referenced from:
      _curses_init in libaa.a(aacurses.o)
      _curses_init in libaa.a(aacurkbd.o)
  "_intrflush", referenced from:
      _curses_init in libaa.a(aacurses.o)
      _curses_uninit in libaa.a(aacurses.o)
      _curses_getsize in libaa.a(aacurses.o)
      _curses_uninit in libaa.a(aacurkbd.o)
  "_keypad", referenced from:
      _curses_init in libaa.a(aacurkbd.o)
      _curses_uninit in libaa.a(aacurkbd.o)
  "_nl", referenced from:
      _curses_uninit in libaa.a(aacurkbd.o)
  "_nocbreak", referenced from:
      _curses_uninit in libaa.a(aacurkbd.o)
  "_nodelay", referenced from:
      _curses_uninit in libaa.a(aacurkbd.o)
      _curses_getchar in libaa.a(aacurkbd.o)
  "_noecho", referenced from:
      _curses_init in libaa.a(aacurkbd.o)
  "_nonl", referenced from:
      _curses_init in libaa.a(aacurkbd.o)
  "_stdscr", referenced from:
      _curses_init in libaa.a(aacurses.o)
      _curses_uninit in libaa.a(aacurses.o)
      _curses_getsize in libaa.a(aacurses.o)
      _curses_setattr in libaa.a(aacurses.o)
      _curses_print in libaa.a(aacurses.o)
      _curses_gotoxy in libaa.a(aacurses.o)
      _curses_flush in libaa.a(aacurses.o)
      ...
  "_termattrs", referenced from:
      _curses_init in libaa.a(aacurses.o)
  "_waddnstr", referenced from:
      _curses_print in libaa.a(aacurses.o)
  "_wclear", referenced from:
      _curses_uninit in libaa.a(aacurses.o)
      _curses_getsize in libaa.a(aacurses.o)
      _curses_uninit in libaa.a(aacurkbd.o)
  "_wgetch", referenced from:
      _curses_getchar in libaa.a(aacurkbd.o)
  "_wmove", referenced from:
      _curses_gotoxy in libaa.a(aacurses.o)
  "_wrefresh", referenced from:
      _curses_uninit in libaa.a(aacurses.o)
      _curses_getsize in libaa.a(aacurses.o)
      _curses_flush in libaa.a(aacurses.o)
      _curses_uninit in libaa.a(aacurkbd.o)
ld: symbol(s) not found for architecture x86_64
clang-7: error: linker command failed with exit code 1 (use -v to see invocation)
ninja: build stopped: subcommand failed.
```

jtojnar commented:
@GrahamcOfBorg build gst_all_1.gst-editing-services gst_all_1.gst-plugins-base gst_all_1.gst-rtsp-server gst_all_1.gstreamer gst_all_1.gst-libav gst_all_1.gst-plugins-good gst_all_1.gst-vaapi gst_all_1.gstreamermm gst_all_1.gst-plugins-bad gst_all_1.gst-plugins-ugly gst_all_1.gst-validate

jtojnar commented:
@GrahamcOfBorg build gst_all_1.gst-editing-services gst_all_1.gst-plugins-base gst_all_1.gst-rtsp-server gst_all_1.gstreamer gst_all_1.gst-libav gst_all_1.gst-plugins-good gst_all_1.gst-vaapi gst_all_1.gstreamermm gst_all_1.gst-plugins-bad gst_all_1.gst-plugins-ugly gst_all_1.gst-validate

jtojnar commented:
https://github.com/mesonbuild/meson/pull/5980 fixed

```
Traceback (most recent call last):
  File "/nix/store/86fh8srw51c2nnawn5y6flmsj6mz3nvd-meson-0.51.2/lib/python3.7/site-packages/mesonbuild/mesonmain.py", line 127, in run
    return options.run_func(options)
  File "/nix/store/86fh8srw51c2nnawn5y6flmsj6mz3nvd-meson-0.51.2/lib/python3.7/site-packages/mesonbuild/msetup.py", line 241, in run
    app.generate()
  File "/nix/store/86fh8srw51c2nnawn5y6flmsj6mz3nvd-meson-0.51.2/lib/python3.7/site-packages/mesonbuild/msetup.py", line 159, in generate
    self._generate(env)
  File "/nix/store/86fh8srw51c2nnawn5y6flmsj6mz3nvd-meson-0.51.2/lib/python3.7/site-packages/mesonbuild/msetup.py", line 188, in _generate
    intr.run()
  File "/nix/store/86fh8srw51c2nnawn5y6flmsj6mz3nvd-meson-0.51.2/lib/python3.7/site-packages/mesonbuild/interpreter.py", line 3924, in run
    super().run()
  File "/nix/store/86fh8srw51c2nnawn5y6flmsj6mz3nvd-meson-0.51.2/lib/python3.7/site-packages/mesonbuild/interpreterbase.py", line 412, in run
    self.evaluate_codeblock(self.ast, start=1)
  File "/nix/store/86fh8srw51c2nnawn5y6flmsj6mz3nvd-meson-0.51.2/lib/python3.7/site-packages/mesonbuild/interpreterbase.py", line 436, in evaluate_codeblock
    raise e
  File "/nix/store/86fh8srw51c2nnawn5y6flmsj6mz3nvd-meson-0.51.2/lib/python3.7/site-packages/mesonbuild/interpreterbase.py", line 430, in evaluate_codeblock
    self.evaluate_statement(cur)
  File "/nix/store/86fh8srw51c2nnawn5y6flmsj6mz3nvd-meson-0.51.2/lib/python3.7/site-packages/mesonbuild/interpreterbase.py", line 451, in evaluate_statement
    return self.evaluate_if(cur)
  File "/nix/store/86fh8srw51c2nnawn5y6flmsj6mz3nvd-meson-0.51.2/lib/python3.7/site-packages/mesonbuild/interpreterbase.py", line 516, in evaluate_if
    self.evaluate_codeblock(i.block)
  File "/nix/store/86fh8srw51c2nnawn5y6flmsj6mz3nvd-meson-0.51.2/lib/python3.7/site-packages/mesonbuild/interpreterbase.py", line 436, in evaluate_codeblock
    raise e
  File "/nix/store/86fh8srw51c2nnawn5y6flmsj6mz3nvd-meson-0.51.2/lib/python3.7/site-packages/mesonbuild/interpreterbase.py", line 430, in evaluate_codeblock
    self.evaluate_statement(cur)
  File "/nix/store/86fh8srw51c2nnawn5y6flmsj6mz3nvd-meson-0.51.2/lib/python3.7/site-packages/mesonbuild/interpreterbase.py", line 443, in evaluate_statement
    return self.assignment(cur)
  File "/nix/store/86fh8srw51c2nnawn5y6flmsj6mz3nvd-meson-0.51.2/lib/python3.7/site-packages/mesonbuild/interpreterbase.py", line 1053, in assignment
    value = self.evaluate_statement(node.value)
  File "/nix/store/86fh8srw51c2nnawn5y6flmsj6mz3nvd-meson-0.51.2/lib/python3.7/site-packages/mesonbuild/interpreterbase.py", line 441, in evaluate_statement
    return self.function_call(cur)
  File "/nix/store/86fh8srw51c2nnawn5y6flmsj6mz3nvd-meson-0.51.2/lib/python3.7/site-packages/mesonbuild/interpreterbase.py", line 778, in function_call
    return func(node, posargs, kwargs)
  File "/nix/store/86fh8srw51c2nnawn5y6flmsj6mz3nvd-meson-0.51.2/lib/python3.7/site-packages/mesonbuild/interpreterbase.py", line 285, in wrapped
    return f(*wrapped_args, **wrapped_kwargs)
  File "/nix/store/86fh8srw51c2nnawn5y6flmsj6mz3nvd-meson-0.51.2/lib/python3.7/site-packages/mesonbuild/interpreterbase.py", line 285, in wrapped
    return f(*wrapped_args, **wrapped_kwargs)
  File "/nix/store/86fh8srw51c2nnawn5y6flmsj6mz3nvd-meson-0.51.2/lib/python3.7/site-packages/mesonbuild/interpreterbase.py", line 285, in wrapped
    return f(*wrapped_args, **wrapped_kwargs)
  [Previous line repeated 1 more time]
  File "/nix/store/86fh8srw51c2nnawn5y6flmsj6mz3nvd-meson-0.51.2/lib/python3.7/site-packages/mesonbuild/interpreterbase.py", line 155, in wrapped
    ret = f(*wrapped_args, **wrapped_kwargs)
  File "/nix/store/86fh8srw51c2nnawn5y6flmsj6mz3nvd-meson-0.51.2/lib/python3.7/site-packages/mesonbuild/interpreterbase.py", line 174, in wrapped
    return f(*wrapped_args, **wrapped_kwargs)
  File "/nix/store/86fh8srw51c2nnawn5y6flmsj6mz3nvd-meson-0.51.2/lib/python3.7/site-packages/mesonbuild/interpreter.py", line 3035, in func_dependency
    d = self.dependency_impl(name, display_name, kwargs)
  File "/nix/store/86fh8srw51c2nnawn5y6flmsj6mz3nvd-meson-0.51.2/lib/python3.7/site-packages/mesonbuild/interpreter.py", line 3082, in dependency_impl
    dep = dependencies.find_external_dependency(name, self.environment, kwargs)
  File "/nix/store/86fh8srw51c2nnawn5y6flmsj6mz3nvd-meson-0.51.2/lib/python3.7/site-packages/mesonbuild/dependencies/base.py", line 2526, in find_external_dependency
    d = c()
  File "/nix/store/86fh8srw51c2nnawn5y6flmsj6mz3nvd-meson-0.51.2/lib/python3.7/site-packages/mesonbuild/dependencies/base.py", line 2389, in __init__
    self.detect(name, paths)
  File "/nix/store/86fh8srw51c2nnawn5y6flmsj6mz3nvd-meson-0.51.2/lib/python3.7/site-packages/mesonbuild/dependencies/base.py", line 2420, in detect
    incdir = self._get_framework_include_path(framework_path)
  File "/nix/store/86fh8srw51c2nnawn5y6flmsj6mz3nvd-meson-0.51.2/lib/python3.7/site-packages/mesonbuild/dependencies/base.py", line 2449, in _get_framework_include_path
    self._get_framework_latest_version(path))
  File "/nix/store/86fh8srw51c2nnawn5y6flmsj6mz3nvd-meson-0.51.2/lib/python3.7/site-packages/mesonbuild/dependencies/base.py", line 2441, in _get_framework_latest_version
    return 'Versions/{}/Headers'.format(sorted(versions)[-1]._s)
IndexError: list index out of range
```

but now there is

```
Traceback (most recent call last):
  File "/nix/store/yi7xpxy8zh02b45aync86mqxi7izs9c8-meson-0.51.2/lib/python3.7/site-packages/mesonbuild/mesonmain.py", line 127, in run
    return options.run_func(options)
  File "/nix/store/yi7xpxy8zh02b45aync86mqxi7izs9c8-meson-0.51.2/lib/python3.7/site-packages/mesonbuild/minstall.py", line 514, in run
    installer.do_install(datafilename)
  File "/nix/store/yi7xpxy8zh02b45aync86mqxi7izs9c8-meson-0.51.2/lib/python3.7/site-packages/mesonbuild/minstall.py", line 343, in do_install
    self.install_targets(d)
  File "/nix/store/yi7xpxy8zh02b45aync86mqxi7izs9c8-meson-0.51.2/lib/python3.7/site-packages/mesonbuild/minstall.py", line 492, in install_targets
    install_name_mappings, verbose=False)
  File "/nix/store/yi7xpxy8zh02b45aync86mqxi7izs9c8-meson-0.51.2/lib/python3.7/site-packages/mesonbuild/scripts/depfixer.py", line 462, in fix_rpath
    fix_darwin(fname, new_rpath, final_path, install_name_mappings)
  File "/nix/store/yi7xpxy8zh02b45aync86mqxi7izs9c8-meson-0.51.2/lib/python3.7/site-packages/mesonbuild/scripts/depfixer.py", line 427, in fix_darwin
    stderr=subprocess.DEVNULL)
  File "/nix/store/9a5bl4gvzdm69fvhwbwln5ld6vn55iyx-python3-3.7.5/lib/python3.7/subprocess.py", line 363, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['install_name_tool', '/nix/store/h3qkfhr1mggkbi962h4gk5vh068vmycr-gst-plugins-bad-1.16.1/lib/gstreamer-1.0/libgstopenexr.dylib', '-id', '/nix/store/h3qkfhr1mggkbi962h4gk5vh068vmycr-gst-plugins-bad-1.16.1/lib/gstreamer-1.0/libgstopenexr.dylib', '-change', '@rpath/libgstopenexr.dylib', '/nix/store/h3qkfhr1mggkbi962h4gk5vh068vmycr-gst-plugins-bad-1.16.1/lib/gstreamer-1.0/libgstopenexr.dylib']' returned non-zero exit status 1.```

nh2 commented:
Works for my use case (SRT-based screen sharing). I ran in 2 terminals:

```
GST_GL_XINITTHREADS=1 NIX_PATH=nixpkgs=. nix-shell -p gst_all_1.gstreamer -p gst_all_1.gst-plugins-good -p gst_all_1.gst-plugins-base -p gst_all_1.gst-plugins-bad -p gst_all_1.gst-plugins-ugly -p gst_all_1.gst-libav --run 'gst-launch-1.0 ximagesrc startx=0 show-pointer=true use-damage=1 ! videoconvert ! x264enc bitrate=3000 tune=zerolatency speed-preset=ultrafast byte-stream=true threads=1 key-int-max=30 intra-refresh=true ! video/x-h264, profile=baseline, framerate=30/1 ! mpegtsmux ! queue ! srtsink uri=srt://127.0.0.1:5000 latency=100'
```

```
NIX_PATH=nixpkgs=. nix-shell -p gst_all_1.gstreamer -p gst_all_1.gst-plugins-good -p gst_all_1.gst-plugins-base -p gst_all_1.gst-plugins-bad -p gst_all_1.gst-plugins-ugly -p gst_all_1.gst-libav --run 'gst-launch-1.0 srtsrc latency=100 uri=srt://:5000/ ! tsdemux ! h264parse ! video/x-h264 ! avdec_h264 ! autovideosink sync=false'
```

For my reference, there are apparently also some SRT related improvements in [this release's changelog](https://gstreamer.freedesktop.org/releases/1.16/#1.16.1):

* srt: Add stats property, include sender-side statistics and fix a crash
* srtserversrc: yields malformed rtp payloads
* srtsink: Fix crash in case no URI
* srt: handle races in state change
* Backport of some minor srt commits without MR into 1.16

nh2 reviewed (approved):
I approve for my use case.

jtojnar committed:
meson: fix darwin framework lookup

Fixes building gst_all_1.gst-plugins-good.

Patch backported from 0.52.0.

jtojnar commented:
Moved the meson fix to meson proper and fully dropped the autotools bit from base.

@NixOS/darwin-maintainers any idea how to fix the [`install_name_tool` issue](https://github.com/NixOS/nixpkgs/pull/70690#issuecomment-553704175) for the `gst_all_1.gst-plugins-bad`?

jtojnar commented:
@GrahamcOfBorg build gst_all_1.gst-editing-services gst_all_1.gst-plugins-base gst_all_1.gst-rtsp-server gst_all_1.gstreamer gst_all_1.gst-libav gst_all_1.gst-plugins-good gst_all_1.gst-vaapi gst_all_1.gstreamermm gst_all_1.gst-plugins-bad gst_all_1.gst-plugins-ugly gst_all_1.gst-validate

jtojnar commented:
The failures appear to be unrelated. Let’s merge and see.

dtzWill commented:
Thanks!
